lib/publiccloud/k8sbasetest.pm

```diff
@@ -22,13 +22,19 @@ Prepare the provider and install kubectl
 =cut
 
 sub init {
-    my ($self) = @_;
+    my ($self, $k8s_backend) = @_;
 
     $self->select_serial_terminal;
     install_kubectl();
 
-    my $service = $self->get_k8s_service_name();
-    $self->provider_factory(service => $service);
+    if (defined $k8s_backend) {
```

Comment:
Not sure what is best practice in perl, but you can just do `if ($k8s_backend)`

Reply:
done